.github/workflows/release.yml

```diff
@@ -21,6 +21,26 @@ jobs:
           bump-minor-pre-major: true
           version-file: "judoscale-ruby/lib/judoscale/version.rb"
 
+      - name: Sync versions - checkout code
+        if: ${{ steps.release.outputs.release_created }}
+        uses: actions/checkout@v4
```

Comment:
I wonder if we'll have to do anything here to checkout the newly-created release PR branch. The action (workflow?) itself is running on `main`, so my guess is that the checkout and commit will also happen on `main`.

In your testing, I think the action was running on the branch, so the automated commit was also on the branch.

If I'm right about this, then maybe we'll want a separate action that only triggers on release PR's, and all that action does is sync the versions.

Not sure how to test any of this, though, without pushing to `main` and creating a release PR.

Reply:
Yeah you're right, I didn't think this through with the different branch it runs vs the one it creates. It couldn't be that simple 😅 

It seems the ["outputs" object has access to a PR object](https://github.com/google-github-actions/release-please-action?tab=readme-ov-file#outputs) if one was created (which it'd have been in this case), and that PR object has access to both head & base branch names: https://github.com/googleapis/release-please/blob/cb106cd0b5ae5c3e2a3b015487dc615eba72b842/src/pull-request.ts#L15-L17, so maybe we can checkout the right branch for that PR and run the sync/commit there? 

I think that'd mean set the `ref` attribute according to [actions/checkout docs](https://github.com/actions/checkout?tab=readme-ov-file#usage), to something like `steps.release.outputs.pr.headBranchName`. It's definitely hard to test all this, without triggering the real thing!

Reply:
I don't think that's controversial at all! It sounds much more reasonable than what we're doing now. 😂

Let's do that!!

Reply:
Sounds good, let's give it a shot.

I pushed a separate PR with that change https://github.com/judoscale/judoscale-ruby/pull/210, since I was wrapping up some other gemspec improvements already, so pushed a PR on top of that to avoid conflicts. Will close this one for now.